lvsfunc/misc.py

```diff
@@ -469,6 +472,123 @@ def colored_clips(amount: int,
+        range_in = Range.LIMITED if clip.format.color_family == vs.ColorFamily.YUV else Range.FULL
+        clips[name] = depth(clip, 8, range_in=range_in, dither_type=dither_type)
+        if (subsampling := get_subsampling(clips[name])) and subsampling not in ('444', None):
+            clips[name] = clips[name].resize.Bicubic(filter_param_a_uv=1/3,
+                                                     filter_param_b_uv=1/3,
+                                                     format=clip.format.replace(subsampling_w=0, subsampling_h=0)
+                                                     )
+
+    if folder:
+        for name, clip in clips.items():
+            if clip.format.color_family == vs.ColorFamily.YUV:
+                matrix_in_s = '709'
+            elif clip.format.color_family == vs.ColorFamily.RGB:
+                matrix_in_s = 'rgb'
+            else:
+                raise ValueError("save: expected a YUV or RGB clip")
+            os.makedirs(name, exist_ok=True)
+            with _cd(name):
+                for f in frames:
+                    out = core.imwri.Write(clip[f].resize.Point(width=(zoom * clip.width),
+                                                                height=(zoom * clip.height),
+                                                                format=vs.RGB24,
+                                                                range=Range.FULL,
+                                                                matrix_in_s=matrix_in_s,
+                                                                dither_type=Dither.ERROR_DIFFUSION.value,
+                                                                prefer_props=True
+                                                                ),
+                                           'PNG',
+                                           '%06d.png',
+                                           firstnum=f
+                                           )
+                    out.get_frame(0)
+    else:
+        for name, clip in clips.items():
+            if clip.format.color_family == vs.ColorFamily.YUV:
+                matrix_in_s = '709'
+            elif clip.format.color_family == vs.ColorFamily.RGB:
+                matrix_in_s = 'rgb'
+            else:
+                raise ValueError("save: expected a YUV or RGB clip")
+            for f in frames:
+                out = core.imwri.Write(clip[f].resize.Point(width=(zoom * clip.width),
+                                                            height=(zoom * clip.height),
+                                                            format=vs.RGB24,
+                                                            range=Range.FULL,
+                                                            matrix_in_s=matrix_in_s,
+                                                            dither_type=Dither.ERROR_DIFFUSION.value,
+                                                            prefer_props=True),
```

Comment:
If a single parameter of a 4-parameter function call spans 7 lines, you *usually* want to declare a variable for the parameter first.

Also, this entire branch is *very* similar to the `folder==True` branch. You should try to unify them and at least share some common code.